echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EventContent.java

```diff
@@ -44,25 +53,51 @@ public EventContent(Event event, String type) throws FieldNotFoundException {
   }
 
   private void setRepo() throws FieldNotFoundException {
-    repo =
+    Map trigger =
         Optional.ofNullable(event.getContent())
             .map(content -> (Map) content.get("execution"))
             .map(execution -> (Map) execution.get("trigger"))
-            .map(trigger -> (Map) trigger.get("buildInfo"))
-            .map(buildInfo -> (String) buildInfo.get("name"))
             .orElseThrow(FieldNotFoundException::new);
+    if (trigger.containsKey("type")
+        && trigger.get("type").equals("git")
+        && trigger.get("source").equals("github")) {
```

Comment:
I would rather do these tests the other way around to protect against potential NPEs:

    "github".equals(trigger.get("source"))